src/dialogs/more-info/controls/more-info-vacuum.ts

```diff
@@ -0,0 +1,286 @@
+  VACUUM_SUPPORT_PAUSE,
+  VACUUM_SUPPORT_RETURN_HOME,
+  VACUUM_SUPPORT_START,
+  VACUUM_SUPPORT_STATUS,
+  VACUUM_SUPPORT_STOP,
+  VacuumEntity,
+} from "../../../data/vacuum";
+
+class VacuumCommand {
+  public translationKey: string;
+  public icon: string;
+  public serviceName: string;
+  private funcIsVisible: (stateObj: VacuumEntity) => boolean;
+
+  constructor(
+    translationKey: string,
+    icon: string,
+    serviceName: string,
+    funcIsVisible: (stateObj: VacuumEntity) => boolean
+  ) {
+    this.translationKey = translationKey;
+    this.icon = icon;
+    this.serviceName = serviceName;
+    this.funcIsVisible = funcIsVisible;
+  }
+
+  public shouldVisible(stateObj: VacuumEntity): boolean {
+    return this.funcIsVisible(stateObj);
+  }
+}
+
+const VACUUM_COMMANDS: VacuumCommand[] = [
+  new VacuumCommand("start", "hass:play", "start", (stateObj) =>
+    supportsFeature(stateObj, VACUUM_SUPPORT_START)
+  ),
+  new VacuumCommand(
+    "pause",
+    "hass:pause",
+    "pause",
+    (stateObj) =>
+      // We need also to check if Start is supported because if not we show play-pause
+      supportsFeature(stateObj, VACUUM_SUPPORT_START) &&
+      supportsFeature(stateObj, VACUUM_SUPPORT_PAUSE)
+  ),
+  new VacuumCommand(
+    "pause",
+    "hass:play-pause",
+    "start_pause",
+    (stateObj) =>
+      // If start is supoorted, we don't show this button
+      !supportsFeature(stateObj, VACUUM_SUPPORT_START) &&
+      supportsFeature(stateObj, VACUUM_SUPPORT_PAUSE)
+  ),
+  new VacuumCommand("stop", "hass:stop", "stop", (stateObj) =>
+    supportsFeature(stateObj, VACUUM_SUPPORT_STOP)
+  ),
+  new VacuumCommand("clean_spot", "hass:broom", "clean_spot", (stateObj) =>
+    supportsFeature(stateObj, VACUUM_SUPPORT_CLEAN_SPOT)
+  ),
+  new VacuumCommand("locate", "hass:map-marker", "locate", (stateObj) =>
+    supportsFeature(stateObj, VACUUM_SUPPORT_LOCATE)
+  ),
+  new VacuumCommand(
+    "return_home",
+    "hass:home-map-marker",
+    "return_to_base",
+    (stateObj) => supportsFeature(stateObj, VACUUM_SUPPORT_RETURN_HOME)
+  ),
+];
+
+// tslint:disable-next-line:max-classes-per-file
+@customElement("more-info-vacuum")
+class MoreInfoVacuum extends LitElement {
+  @property() public hass!: HomeAssistant;
+
+  @property() public stateObj?: VacuumEntity;
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.stateObj) {
+      return html``;
+    }
+
+    const stateObj = this.stateObj;
+
+    const supportsPause = supportsFeature(stateObj, VACUUM_SUPPORT_PAUSE);
+    const supportsStop = supportsFeature(stateObj, VACUUM_SUPPORT_STOP);
+    const supportsReturnHome = supportsFeature(
+      stateObj,
+      VACUUM_SUPPORT_RETURN_HOME
+    );
+    const supportsFanSpeed = supportsFeature(
+      stateObj,
+      VACUUM_SUPPORT_FAN_SPEED
+    );
+    const supportsBattery = supportsFeature(stateObj, VACUUM_SUPPORT_BATTERY);
+    const supportsStatus = supportsFeature(stateObj, VACUUM_SUPPORT_STATUS);
```

Comment:
Since we use these only once, it is nice to inline the check instead of creating a const.

Reply:
Changed it to inline, except for `supportsCommandBar` as in my opinion it is clearer if it is a variable and not checked inline.

Reply:
Actually, I only meant the ones that were used once, the ones that are used twice (in `supportsCommandBar`) should be a `const`.

Reply:
Today I had another idea, I can use `VACUUM_COMMANDS.some(...)` so we don't have any duplicate code and the maintenance of the code is easier.